aptos-move/aptos-vm/src/move_vm_ext/resolver.rs

```diff
@@ -66,39 +77,18 @@ pub trait MoveResolverExt:
 
     fn get_resource_group(&self, struct_tag: &StructTag) -> Result<Option<StructTag>, VMError> {
         let metadata = self.get_module_metadata(struct_tag.module_id());
-        Ok(Self::get_resource_group_from_metadata(struct_tag, metadata))
-    }
-
-    fn get_resource_group_from_metadata(
-        struct_tag: &StructTag,
-        metadata: Option<aptos_framework::RuntimeModuleMetadataV1>,
-    ) -> Option<StructTag> {
-        metadata.and_then(|metadata| {
-            metadata
-                .struct_attributes
-                .get(struct_tag.name.as_ident_str().as_str())
-                .and_then(|attrs| {
-                    attrs
-                        .iter()
-                        .find_map(|attr| attr.get_resource_group_member())
-                })
-        })
+        Ok(get_resource_group_from_metadata(struct_tag, metadata))
     }
 
     fn is_resource_group(&self, struct_tag: &StructTag) -> bool {
-        let metadata = self.get_module_metadata(struct_tag.module_id());
-        metadata
-            .and_then(|metadata| {
-                metadata
-                    .struct_attributes
-                    .get(struct_tag.name.as_ident_str().as_str())
-                    .and_then(|attrs| {
-                        attrs
-                            .iter()
-                            .map(|attr| Some(attr.is_resource_group()))
-                            .next()
-                    })
-            })
-            .is_some()
+        (|| {
```

Comment:
this closure looks weird to me, why do we need it?

Reply:
To use ? chaining, instead of the IMO awkward and_then(closure).

Reply:
i see, and_then is probably the idiomatic way of doing this but I agree it looks ugly

Reply:
both are pretty uncomfortable... the new way superficially reads cleaner but demonstrates a much higher degree of mastery of the language than the former.

Given that both Zekun and I are unaware of this power, it likely means we're writing code that few others would understand. I tend to err toward writing code that can be commonly understood.

On the flip side, the new method is a lot tighter and maybe we just haven't had any new blood in rust for a long time hence our skills have stagnated.

Reply:
+1 to chaining (e.g., `and_then`). Even though `and_then` is more verbose, it took me 10x longer to figure out what was going on here 😵‍💫 

> it likely means we're writing code that few others would understand.

Yeah, I like simple code so that simple people (like me) can still have a job 😄

Reply:
I added aptos_try as a generic utility, in lieu of try language feature in unstable.

I think it's a useful pattern as I've seen this more in the code